src/anemoi/inference/outputs/netcdf.py

```diff
@@ -47,7 +48,8 @@ def _init(self, state):
         if os.path.exists(self.path):
             os.remove(self.path)
 
-        self.ncfile = Dataset(self.path, "w", format="NETCDF4")
+        with LOCK:
```

Comment:
When testing `anemoi-inference run` with netcdf output, it hangs forever on `with LOCK:`  (Not even proceeding to the next line). Replacing this with `if True:` the code proceeds to hang on the next instance of `with LOCK`. Possibly this is related to something missing in my installation. But no error messages or warnings appear.

I suggest the use of `with LOCK` is checked, in case it works in certain instances then and least an `assert` statement, making sure it will work or when not throw a message, needs to be added.

Reply:
Ok, I'll have a look. Maybe we need an RLock

Reply:
Indeed, that solves the problem.